.github/workflows/build-publish-develop-pr.yml

```diff
@@ -19,6 +19,36 @@ env:
   GIT_REF: ${{ github.event.inputs.git_ref || github.ref }}
 
 jobs:
+  changes:
+    name: Find changed files
+    runs-on: ubuntu-latest
+    outputs:
+      not_ignored_changes: ${{ steps.filter.outputs.every_changed == 'true' || steps.filter.outputs.any_changed == 'true' }}
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+        with:
+          ref: ${{ env.GIT_REF }}      
+      - name: Filter changes
+        uses: ./.github/actions/filter-changes
+        id: filter
+        with:
+          # Filter files that match **every** pattern in this list
+          every-filter: |
+            - '**'                     # Initially include all files
+            - '!**/*.md'               # Exclude all markdown files (lowercase)
+            - '!**/*.MD'               # Exclude all markdown files (uppercase)
+            - '!**/.github/**'         # Exclude all changes under the .github directory
+            # Exclude all Go test files, except those explicitly included in "any-filter"
+            - '!**/*_test.go'
+          # Filter files that match **any** of these patterns
+          any-filter: |
+            - 'integration-tests/**/*.go'
```

Comment:
Given the complexity / sheer number of filters we have, it'd be reasonable to create a fork of https://github.com/dorny/paths-filter/blob/master/src/main.ts which would allow us to actually have a test suite. 

This way reviewers can easily see which types of file changes would trigger changes, and which ones would not. Otherwise, these filters are going to fail overtime as they are too easy to forget to update

Reply:
@HenryNguyen5 I created https://smartcontract-it.atlassian.net/browse/TT-1809. cc @AKhozya